lib/cosmos/dart/lib/dart_packet_log_writer.rb

```diff
@@ -0,0 +1,103 @@
+require File.expand_path('../../config/environment', __FILE__)
+require 'dart_common'
+
+class DartPacketLogWriter < Cosmos::PacketLogWriter
+  include DartCommon
+
+  DEFAULT_SYNC_COUNT_LIMIT = 100
+
+  def initialize(*args)
+    super(*args)
+    @packet_log_id = nil
+    @meta_id = nil
+    @db_queue = Queue.new
+    @sync_count = 0
+    @sync_count_limit = DEFAULT_SYNC_COUNT_LIMIT
+    @not_ready_ple_ids = []
+
+    sync_targets_and_packets()
+
+    @db_thread = Cosmos.safe_thread("Database packet log entries") do
+      db_thread_body()
+    end
+  end
+
+  def shutdown
+    super()
+    Cosmos.kill_thread(self, @db_thread)
+  end
+
+  def graceful_kill
+    super()
+    @db_queue << nil
+  end
+
+  protected
+
+  def start_new_file_hook(packet)
+    packet_log = PacketLog.new
+    packet_log.filename = @filename.clone
```

Comment:
This is recording the absolute path for the filename. Does that cause problems if someone changes the DART_DATA environment variable? Seems like it would have to re-import everything.

Reply:
My cleanup code removes information from the database if files go missing.  That may be overkill if all you have to do is remove an environment variable to destroy the raw data in the database.   The DART_DATA value is new and moving to relative paths would probably be better at this point.   I'll make this change.

Reply:
Rethinking this I think the absolute paths are safer.  Is there any legitimate reason why the log files would move?   If they need to be moved then we should add a database command to do that.

Reply:
They'd change because someone likes a new path better: "Data" vs "Archive" or something stupid like that. The main thing is that you wouldn't want the entire database to have to be re-built because of something like that. I think the safest thing is to abort the entire process with a warning message if that happens. Perhaps a command line option like --force to rebuilt the whole thing and --relocate to note a new directory.